plugins/catalog-backend/src/ingestion/processors/ldap/client.ts

```diff
@@ -18,12 +18,24 @@ import ldap, { Client, SearchEntry, SearchOptions } from 'ldapjs';
 import { BindConfig } from './config';
 import { errorString } from './util';
 
+/**
+ * A string representation of a Microsoft Active Directory Server.
+ */
+export const LDAP_VENDOR_ACTIVE_DIRECTORY = 'Microsoft Active Directory';
```

Comment:
So i was thinking about this one.

First thought was, at least this should be just

```ts
export LdapVendor = 'microsoft-active-directory' | 'unknown';
```

and that might be a good change on its own.

But then I thought, maybe the vendor type could rather be encapsulating what the nuances are of that particular vendor.

```ts
type LdapVendor = {
  dnAttributeName: string;
  // decodes only certain ones, on MS AD
  decodeAttribute: (name: string, value: string | Buffer | Buffer[]) => string;
  // more?
}
```

and then the default vendor is whatever leads to the original code behaving just like it did before.

Reply:
Thanks for the advice. I went with your second suggestion and have isolated the nuances between the vendors.